packages/guides/src/ReferenceResolvers/DelegatingReferenceResolver.php

```diff
@@ -20,21 +21,25 @@ public function __construct(private readonly iterable $resolvers, private readon
     {
     }
 
-    public function resolve(LinkInlineNode $node, RenderContext $renderContext): void
+    public function resolve(LinkInlineNode $node, RenderContext $renderContext, Messages|null $messages = null): void
     {
+        if ($messages === null) {
+            $messages = new Messages();
+        }
```

Comment:
```suggestion
        $messages ??= new Messages();
```

Reply:
the DelegatingReferenceResolver is calling all the other resolvers and takes care of the warnings. However it  implements the same Interface. We could however move the message handling to ReferenceResolverPreRender...

Reply:
I moved the Message handling to the PreRenderer. Please have another look